wgpu-core/src/command/bundle.rs

```diff
@@ -667,13 +667,81 @@ pub enum RenderBundleError {
     Draw(#[from] DrawError),
 }
 
+/// Error encountered when performing a render pass.
+#[derive(Clone, Debug, Error)]
+pub enum RenderBundleError {
+    #[error(transparent)]
+    Inner(#[from] RenderBundleErrorInner),
```

Comment:
Honestly I just did it like this to make use of `thiserror`s `#[error]` annotations in a simple way, as I was under the impression that I would have to write the Error implementations manually if I make it a regular struct - I might be wrong about that though. In any case I thought that it doesnt matter much as it won't take up more or less space in the types memory layout anyway.

Reply:
Actually nevermind that, just re-read the docs of thiserror, you can use it with structs as well of course. Then that leaves just the issue that I'm not sure it would be relevantly shorter if I re-write as a struct (and it would also lose the functionality of the `Inner` variant, which does _not_ attach an error context and as such can attach the error messages direct to the RenderCommand itself)

I'll prototype it though, lets see how it will look.

Reply:
How about this?
```rust

fn error_context(command: ComputeCommand) -> ComputePassCommandError {
    match command {
        ComputeCommand::SetBindGroup { .. } => ComputePassCommandError::SetBindGroup,
        ComputeCommand::SetPipeline(_) => ComputePassCommandError::SetPipeline,
        ComputeCommand::SetPushConstant { .. } => ComputePassCommandError::SetPushConstant,
        ComputeCommand::Dispatch(_) => ComputePassCommandError::Dispatch,
        ComputeCommand::DispatchIndirect { .. } => ComputePassCommandError::DispatchIndirect,
        ComputeCommand::PushDebugGroup { .. }
        | ComputeCommand::PopDebugGroup
        | ComputeCommand::InsertDebugMarker { .. } => ComputePassCommandError::Debug,
    }
}

/// Error encountered when performing a compute pass.
#[derive(Clone, Debug, Error)]
#[error("{command}")]
pub struct ComputePassError {
    command: ComputePassCommandError,
    #[source]
    inner: ComputePassErrorInner,
}

#[derive(Clone, Debug, Error)]
pub enum ComputePassCommandError {
    #[error("In the pass parameters")]
    Pass,
    #[error("In a set_bind_group command")]
    SetBindGroup,
    #[error("In a set_pipeline command")]
    SetPipeline,
    #[error("In a set_push_constant command")]
    SetPushConstant,
    #[error("In a dispatch command")]
    Dispatch,
    #[error("In a indirect dispatch command")]
    DispatchIndirect,
    #[error("In a debug command")]
    Debug,
}
```

I could store the `RenderCommand` directly, but then I would still need a match to map to the error messages - it would remove the enum I suposed, but it seems kinda unclean to carry the payload of the commands around if we only care about the descriminant.

Reply:
I could also make it shorter by storing the error message as a field:

```rust
type ComputePassCommandError = Option<&'static str>;

fn error_context(command: ComputeCommand) -> ComputePassCommandError {
    Some(match command {
        ComputeCommand::SetBindGroup { .. } => "In a set_bind_group command",
        ComputeCommand::SetPipeline(_) => "In a set_pipeline command",
        ComputeCommand::SetPushConstant { .. } => "In a set_push_constant command",
        ComputeCommand::Dispatch(_) => "In a dispatch command",
        ComputeCommand::DispatchIndirect { .. } => "In a indirect dispatch command",
        ComputeCommand::PushDebugGroup { .. }
        | ComputeCommand::PopDebugGroup
        | ComputeCommand::InsertDebugMarker { .. } => "In a debug command",
    })
}

/// Error encountered when performing a compute pass.
#[derive(Clone, Debug, Error)]
#[error("{command}")]
pub struct ComputePassError {
    command: &'static str,
    #[source]
    inner: ComputePassErrorInner,
}

impl ComputePassError {
    fn new(command: ComputePassCommandError, inner: ComputePassErrorInner) -> Self {
        ComputePassError {
            command: command.unwrap_or("In the pass parameters"),
            inner,
        }
    }
}
```

Reply:
The last one looks good to me, thank you for experimenting!
I was thinking that just debug-printing `RenderCommand` would be simple and effective, too, since it would avoid the need to remap it into anything. But your last `error_context` looks simple enough.